### ISSUE TITLE
Does this file 'id_paths.json' contain the test dataset?

### ISSUE THREAD

**Issue by 929343413** [YicongHong/Recurrent-VLN-BERT#16]
When I run on the R2R_test dataset, the nDTW will have a key error. So I guess the id_paths.json doesn't contain the R2R_test dataset, is it  right?

**Comment by YicongHong** [comment]
Hi There,

Yes, that's right. The R2R test set is reserved for fair testing, so the ground-truth paths are not publicly available.

Cheers

**Comment by 929343413** [comment]
Thanks for your reply!

But I have a new question now. May I ask whether the relationship between batch_size, epoch and iteration in this experiment is different from that in the traditional experiment. In your code, epoch seems to be set with iters and it is set to 300000 times. In addition, batch_size and iteration seem to be set separately, and iteration you seem to set like this ' interval = min(log_every, n_iters-idx) '.I don't understand why it is set like this.
In my view, in the traditional training method, batch size is the number of sample data in a batch, so the number of batches is to divide a complete dataset into several batches, and the number of batches is the number of iterations, right? So I'm a bit confused here.
  
Sorry to bother you, I'm still new in this field.

**Comment by YicongHong** [comment]
Hi, 
Sorry for my late reply.
The code follows the codebase of the [EnvDrop](https://github.com/airsplay/R2R-EnvDrop), it doesn't define 'epoch', but simply use number of iterations to indicate number of updates of the network.
The for loop in [train.py](https://github.com/YicongHong/Recurrent-VLN-BERT/blob/main/r2r_src/train.py) is the log loop, which controls how many iterations done (interval) before running a validation and logging the statistics. The for loop in [agent.py](https://github.com/YicongHong/Recurrent-VLN-BERT/blob/main/r2r_src/agent.py) is the actual iteration loop.
Note that here we follow EnvDrop to do IL+RL training, each iteration will rollout two navigations -- one for IL (self.feedback = 'teacher') and one for RL (self.feedback = 'sample'). So, by setting batch size to N, the code actually runs 2*N samples each iterations.
Hope this helps.

**Comment by 929343413** [comment]
Thanks for your reply! 
Could you tell me why you directly use the model pretrained by PREVALENT for navigation? And do the pre-training tasks, AP and MLM, also work in the training?
Besides, is the goal of Recurrent VLN BERT for pretraining? Now I use it in training, does it have a better performance as well?
Looking forward to your reply.Thanks.

**Comment by YicongHong** [comment]
Sorry for my late reply, hopefully you have found the answers. 
Othewise, I believe we have answered all your questions in the paper and explained why.
Thanks!
